### PR TITLE
Random sample

### PR DESCRIPTION
**Pull request by smallfishabc**
Hi Steph,
This is my random sample branch with the ensemble generation ability. Could you have a look at this? I checked and there is not much overlap or conflicts with the main branch.
Best
Feng Yu

### CODE REVIEW

File: 1ael.pdb
Comment (stephprince) [inline 1ael.pdb:1]: I don't think this file should be included

File: scripts/run_ensemble_model/DEBUG_README.md
Comment (stephprince) [inline scripts/run_ensemble_model/DEBUG_README.md:1]: I think this and `debug_train_structure.sh` and `test_imports.py` are leftovers from an AI debugging run so could be removed but if you are still using them feel free to leave

File: src/metfish/msa_model/data/data_modules.py [hunk 1]
```diff
@@ -11,16 +11,12 @@ class MSASAXSDataset(Dataset):
...
-               pdb_dir=None, pdb_prefix=None, pdb_ext=None,
                msa_dir=None,
                saxs_dir=None, saxs_ext=None):
       self.pdb_chains = pd.read_csv(path, index_col='name')#.sort_index()
       self.mode = 'train' if mode is None else 'predict'
       self.data_dir = data_dir
       self.msa_dir = msa_dir
-      self.pdb_dir = pdb_dir
-      self.pdb_prefix = pdb_prefix if pdb_prefix is not None else "fixed_"
-      self.pdb_ext = pdb_ext or ".pdb"
```
Comment (stephprince) [inline src/metfish/msa_model/data/data_modules.py:23]: I think the changes in this file would break some of the current training code - could you adapt your training code to use these args and uncomment the code below

File: src/metfish/msa_model/utils/loss.py [hunk 2]
```diff
@@ -1627,12 +1628,12 @@ def saxs_loss(all_atom_pred_pos: torch.Tensor,
     if use_l1:
         # calculate the L1 divergence of the SAXS profiles
         l1_loss = nn.L1Loss(reduction="sum")
-        loss = l1_loss(pred_saxs, true_saxs)
+        loss = l1_loss(pred_saxs, saxs)
```
Comment (stephprince) [inline src/metfish/msa_model/utils/loss.py:1702]: could you revert these changes? I don't think saxs should be an input anymore in this function

File: pyproject.toml [hunk 3]
```diff
@@ -52,6 +52,7 @@ training = ["wandb==0.16.6",
             "openfold @ git+https://github.com/aqlaboratory/openfold.git@103d037",
             "ml-collections==0.1.0",
             "dm-tree==0.1.6",
+            "lightning==2.1.0",
```
Comment (stephprince) [inline pyproject.toml:55]:
```suggestion
```

This is already specified above

**Review by stephprince — COMMENTED** [review]
Added my comments! I was mainly reviewing for any changes that would break the core model training so I did not look in detail at the `refinement_model` module updates
